Adds SSL support

keaner opened the pull request:
Adds serveSecure method which passes through to bindSecure

kevmoo commented:
I like it – although I wonder if we should just add `securityContext` as an optional argument to serve.

@nex3 ?

jakemac53 commented on lib/shelf_io.dart:
```diff
@@ -57,6 +57,23 @@ void serveRequests(Stream<HttpRequest> requests, Handler handler) {
+Future<HttpServer> serveSecure(
+    Handler handler, address, int port, SecurityContext serverContext,
+    {int backlog}) {
+  if (backlog == null) backlog = 0;
```
nit: `backlog ??= 0`

nex3 commented:
> I like it – although I wonder if we should just add `securityContext` as an optional argument to serve.

I like this idea.

keaner commented:
Cool, I'll make SecurityContext an optional arg and add tests

kevmoo commented:
Awesome.

keaner commented:
Updated based on recommendations and added two tests.

I originally tried to add the certs as separate files but I was running into this error:
"Unsupported operation: Cannot extract a file path from a data URI"

nex3 commented on lib/shelf_io.dart:
```diff
@@ -30,11 +30,22 @@ export 'src/io_server.dart';
+    {SecurityContext securityContext, int backlog}) {
+  backlog ??= 0;
+  if (securityContext == null) {
+    return HttpServer.bind(address, port, backlog: backlog).then((server) {
```
Use async/await here rather than duplicating `then()` calls.

keaner replied:
Done!

nex3 commented on test/shelf_io_ssl_test.dart:
```diff
@@ -0,0 +1,69 @@
+  return schedule/*<Future<HttpClientRequest>>*/(() {
+    return client.getUrl(Uri.parse('https://localhost:$_serverPort/'));
+  });
+}
```
Can you make this part of `shelf_io_test.dart`?

keaner replied:
Done!
